planet/clients/subscriptions.py

```diff
@@ -263,3 +267,45 @@ class _ResultsPager(Paged):
             raise
         except ClientError:  # pragma: no cover
             raise
+
+    async def get_results_csv(self,
+                              subscription_id: str,
+                              status: Optional[Sequence[Literal[
+                                  "created",
+                                  "queued",
+                                  "processing",
+                                  "failed",
+                                  "success"]]] = None,
+                              **kwargs) -> AsyncIterator[str]:
+        """Iterate over rows of results CSV for a Subscription.
+
+        Notes:
+            The name of this method is based on the API's method name. This
+            method provides iteration over results, it does not get a
+            single result description or return a list of descriptions.
+
+        Parameters:
+            subscription_id (str): id of a subscription.
+            status (Set[str]): pass result with status in this set,
+                filter out results with status not in this set.
+            TODO: created, updated, completed, user_id
+
+        Yields:
+            str: a row from a CSV file.
+
+        Raises:
+            APIError: on an API server error.
+            ClientError: on a client error.
+        """
+        url = f'{self._base_url}/{subscription_id}/results'
+        params = {'status': [val for val in status or {}], 'format': 'csv'}
+
+        # Note: retries are not implemented yet. This project has
+        # retry logic for HTTP requests, but does not handle errors
+        # during streaming. We may want to consider a retry decorator
+        # for this entire method a la stamina:
+        # https://github.com/hynek/stamina.
```

Comment:
I think that our model for retrying requests doesn't cover streaming cases cases.